tiledb/sm/filesystem/s3.cc

```diff
@@ -596,6 +596,49 @@ Status S3::move_dir(const URI& old_uri, const URI& new_uri) {
   return Status::Ok();
 }
 
+Status S3::copy_file(const URI& old_uri, const URI& new_uri) {
+  RETURN_NOT_OK(init_client());
+
+  RETURN_NOT_OK(copy_object(old_uri, new_uri));
+  return Status::Ok();
+}
+
+Status S3::copy_dir(const URI& old_uri, const URI& new_uri) {
+  RETURN_NOT_OK(init_client());
+
+  std::cerr << "TEST: " << std::endl;
+  std::string old_uri_string = old_uri.to_string();
+
+  std::vector<std::string> paths;
+  RETURN_NOT_OK(ls(old_uri, &paths));
+  while (!paths.empty()) {
+    std::string file_name_abs = paths.front();
+    URI file_name_uri = URI(file_name_abs);
+    std::cerr << "OLD: " << file_name_abs << std::endl;
+    std::string file_name = file_name_abs.substr(old_uri_string.length());
+    paths.erase(paths.begin());
+    bool dir_exists;
+    RETURN_NOT_OK(is_dir(file_name_uri, &dir_exists));
+    if (dir_exists) {
+      std::cerr << "\n";
+      // RETURN_NOT_OK(create_dir(new_path + "/" + file_name));
+      std::vector<std::string> child_paths;
+      RETURN_NOT_OK(ls(file_name_uri, &child_paths));
+      paths.insert(paths.end(), child_paths.begin(), child_paths.end());
+    } else {
+      bool file_exists;
+      RETURN_NOT_OK(is_object(file_name_uri, &file_exists));
```

Comment:
Since this is just a sanity check, let's remove it. This makes an HTTP request which is too expensive since we don't need it. We'll assume that everything that is not a directory is a file.